### PR TITLE
 feat(xref): log error counts

### PR DESCRIPTION
**Pull request by sidvishnoi**
It looks like this:
```
Listening on port 8000!
2019-08-19T09:55:15.809Z | ::1 | POST 200 /xref/ | - | 155 | 20.678 ms | -
2019-08-19T09:55:19.248Z | ::1 | POST 200 /xref/ | - | 251 | 1.585 ms | {"errors":2,"queries":3}
```

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -9,12 +9,16 @@ const app = express();
 const helmet = require("helmet");
 app.use(compression());
 
-
-// loggin
+// logging
+morgan.token("locals", (req, res) => {
```
Comment (marcoscaceres) [inline app.js]: Fancy!

Comment (marcoscaceres) [reply app.js]: I guess this automatically checks for `undefined`

Comment (sidvishnoi) [reply app.js]: Yes. `undefined` becomes `-`

**Comment by sidvishnoi** [comment]
Only issue with this approach is that we can't send these logs to stderr

File: routes/xref/index.js [hunk 2]
```diff
@@ -2,7 +2,24 @@
 const { search } = require("respec-xref-route/search");
 
 module.exports.route = function route(req, res) {
-  const { keys, options } = req.body;
+  const { keys = [], options } = req.body;
   const body = search(keys, options);
+
+  const errors = getErrorCount(body.result);
+  if (errors / keys.length > 0.05) {
+    // add to logs if error rate is more than 5%
```
Comment (sidvishnoi) [inline routes/xref/index.js]: Should we send all to logs instead or 5% rate is good? (or `if (errors.length)`)

Comment (marcoscaceres) [reply routes/xref/index.js]: send all, I think... if it's too noisy, then we can got for 5%

**Pull request merged**